lib/daru/index.rb

```diff
@@ -419,4 +501,109 @@ def nils_counted array
       }
     end
   end
-end
+
+  class CategoricalIndex < Index
+    
+    include Enumerable
```

Comment:
Yes each class index is enumerable to support all the methods such as  `each_with_index`.